src/Cli/dotnet/commands/dotnet-workload/install/NetSdkManagedInstaller.cs

```diff
@@ -0,0 +1,283 @@
+            var workloadManifestProvider = new SdkDirectoryWorkloadManifestProvider(_dotnetDir, featureBand);
+            return WorkloadResolver.Create(workloadManifestProvider, _dotnetDir, featureBand);
+        }
+
+        public override IReadOnlyCollection<string> GetFeatureBandsWithInstallationRecords()
+        {
+            if (Directory.Exists(_workloadMetadataDir))
+            {
+                var bands = Directory.EnumerateDirectories(_workloadMetadataDir);
+                return bands
+                    .Where(band => Directory.Exists(Path.Combine(band, _installedWorkloadDir)))
+                    .Select(path => Path.GetFileName(path))
+                    .ToList().AsReadOnly();
+            }
+            else
+            {
+                return new List<string>().AsReadOnly();
+            }
+        }
+
+        public override IReadOnlyCollection<string> GetInstalledWorkloads(string featureBand)
+        {
+            var path = Path.Combine(_workloadMetadataDir, featureBand, _installedWorkloadDir);
+            if (Directory.Exists(path))
+            {
+                return Directory.EnumerateFiles(path)
+                    .Select(file => Path.GetFileName(file))
+                    .ToList().AsReadOnly();
+            }
+            else
+            {
+                return new List<string>().AsReadOnly();
+            }
+        }
+
+        public override void WriteWorkloadInstallationRecord(string workloadId, string featureBand)
+        {
+            _reporter.WriteLine(string.Format(LocalizableStrings.WritingWorkloadInstallRecordMessage, workloadId));
+            var path = Path.Combine(_workloadMetadataDir, featureBand, _installedWorkloadDir, workloadId);
+            if (!Directory.Exists(Path.GetDirectoryName(path)))
+            {
+                Directory.CreateDirectory(Path.GetDirectoryName(path));
+            }
+            File.WriteAllText(path, string.Empty);
+        }
+
+        public override void DeleteWorkloadInstallationRecord(string workloadId, string featureBand)
+        {
+            var path = Path.Combine(_workloadMetadataDir, featureBand, _installedWorkloadDir, workloadId);
+            if (File.Exists(path))
+            {
+                File.Delete(path);
+            }
+        }
+
+        private bool PackIsInstalled(PackInfo packInfo)
+        {
+            if (packInfo.Kind.Equals(WorkloadPackKind.Library) || packInfo.Kind.Equals(WorkloadPackKind.Template))
+            {
+                return File.Exists(packInfo.Path);
+            }
+            else
+            {
+                return Directory.Exists(packInfo.Path);
+            }
+        }
+
+        private void DeletePack(PackInfo packInfo)
+        {
+            if (PackIsInstalled(packInfo))
+            {
+                if (packInfo.Kind.Equals(WorkloadPackKind.Library) || packInfo.Kind.Equals(WorkloadPackKind.Template))
+                {
+                    File.Delete(packInfo.Path);
```

Comment:
check for existence?

Reply:
It will never happen or there is a missing test?

Reply:
The existence check is done in `PackIsInstalled`, so at this point we know this file exists